groovy-language-server/src/test/java/com/palantir/groovylanguageserver/GroovycWrapperTest.java

```diff
@@ -379,18 +383,589 @@ public void testCompile_error() throws InterruptedException, ExecutionException,
+                        SymbolKind.Variable, -1, -1, -1, -1, Optional.of("setMyFriend"))),
+                wrapper.getTypeReferences().get("Cat2$InnerCat2"));
+        // Find one line enum correctly
+        assertEquals(Sets.newHashSet(
+                createSymbolInformation("ONE", enumFile.getAbsolutePath(),
+                        SymbolKind.Field, 1, 14, 1, 17, Optional.of("MyEnum")),
+                createSymbolInformation("TWO", enumFile.getAbsolutePath(),
+                        SymbolKind.Field, 1, 18, 1, 21, Optional.of("MyEnum"))),
+                wrapper.getTypeReferences().get("MyEnum").stream()
+                        .filter(symbol -> Ranges.isValid(symbol.getLocation().getRange())).collect(Collectors.toSet()));
+        // Identify type A correctly
+        assertEquals(Sets.newHashSet(
+                createSymbolInformation("a", innerClass.getAbsolutePath(),
+                        SymbolKind.Field, 2, 1, 2, 4, Optional.of("A"))),
+                wrapper.getTypeReferences().get("A").stream()
+                        .filter(symbol -> Ranges.isValid(symbol.getLocation().getRange())).collect(Collectors.toSet()));
+        // Identify type B correctly
+        assertEquals(Sets.newHashSet(
+                createSymbolInformation("b", innerClass.getAbsolutePath(),
+                        SymbolKind.Field, 3, 1, 3, 4, Optional.of("A"))),
+                wrapper.getTypeReferences().get("A$B").stream()
+                        .filter(symbol -> Ranges.isValid(symbol.getLocation().getRange())).collect(Collectors.toSet()));
+    }
+
+    @Test
+    public void testFindReferences_edgeCases() throws IOException {
+        File newFolder1 = root.newFolder();
+        // edge cases, intersecting ranges
+        File file = addFileToFolder(newFolder1, "Dog.groovy",
+                "class Dog {\n"
+                + "   Cat friend1;\n"
+                + "   Cat2 friend2;\n"
+                + "   Cat bark(Cat enemy) {\n"
+                + "      println \"Bark! \" + enemy.name\n"
+                + "      return friend1\n"
+                + "   }\n"
+                + "}\n"
+                + "class Cat {\n"
+                + "   public String name = \"Bobby\"\n"
+                + "}\n"
+                + "class Cat2 {\n"
+                + "   InnerCat2 myFriend;\n"
+                + "   class InnerCat2 {\n"
+                + "   }\n"
+                + "}\n");
+        // edge case on one line
+        File enumFile = addFileToFolder(newFolder1, "MyEnum.groovy",
+                "enum MyEnum {ONE,TWO}\n");
+        // edge case on one line
+        File innerClass = addFileToFolder(newFolder1, "AandB.groovy",
+                "public class A {public static class B {}\n"
+                        + "A a\n"
+                        + "B b\n"
+                + "}\n");
+        GroovycWrapper wrapper = GroovycWrapper.of(output.getRoot().toPath(), root.getRoot().toPath());
```

Comment:
The files need to be created before the wrapper is called since it looks through the `workspaceRoot` and adds the sources to the `CompilationUnit`. Also some of the tests need to do things before this is called: `testTargetDirectoryNotFolder` and `testWorkspaceRootNotFolder`.